engine/src/main/java/org/terasology/world/internal/WorldProviderCoreImpl.java

```diff
@@ -333,6 +334,12 @@ public int getExtraData(int index, int x, int y, int z) {
     public int setExtraData(int index, Vector3i worldPos, int value) {
         Vector3i chunkPos = ChunkMath.calcChunkPos(worldPos);
         CoreChunk chunk = chunkProvider.getChunk(chunkPos);
+        return setExtraData(index, chunk, worldPos,value);
+    }
+
+    @Override
+    public int setExtraData(int index, CoreChunk chunk, Vector3i worldPos, int value) {
```

Comment:
By convention, if a method takes a chunk, it should take a relative position, not absolute as here.